File: backend/Nexora.Domain/Entities.cs
using System;
using System.Collections.Generic;

namespace Nexora.Domain.Entities
{
    public class User
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public string Email { get; set; } = string.Empty;
        public string PasswordHash { get; set; } = string.Empty;
        public string FullName { get; set; } = string.Empty;
        public string Role { get; set; } = "Analyst";
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }

    public class Agent
    {
        public string Id { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Role { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Icon { get; set; } = string.Empty;
    }

    public class DebateSession
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public Guid? UserId { get; set; }
        public string Question { get; set; } = string.Empty;
        public string Context { get; set; } = string.Empty;
        public string Status { get; set; } = "Pending"; // Pending, Running, Completed, Failed
        public double? ConsensusScore { get; set; }
        public double? ConfidenceScore { get; set; }
        public double? AgreementLevel { get; set; }
        public int DurationMs { get; set; }
        public string LlmProvider { get; set; } = "OpenAI";
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? CompletedAt { get; set; }

        public List<AgentOpinion> Opinions { get; set; } = new();
        public List<DebateMessage> DebateMessages { get; set; } = new();
        public ConsensusDecision? ConsensusDecision { get; set; }
    }

    public class AgentOpinion
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public Guid DebateSessionId { get; set; }
        public string AgentId { get; set; } = string.Empty;
        public string Recommendation { get; set; } = string.Empty;
        public string ReasoningSummary { get; set; } = string.Empty;
        public double Confidence { get; set; }
        public string RisksJson { get; set; } = "[]";
        public string AssumptionsJson { get; set; } = "[]";
        public string CounterargumentsJson { get; set; } = "[]";
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }

    public class DebateMessage
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public Guid DebateSessionId { get; set; }
        public string SenderAgentId { get; set; } = string.Empty;
        public string? TargetAgentId { get; set; }
        public string MessageType { get; set; } = "Critique"; // Critique, Challenge, Rebuttal, Agreement
        public string Content { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }

    public class ConsensusDecision
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public Guid DebateSessionId { get; set; }
        public string FinalRecommendation { get; set; } = string.Empty;
        public string Summary { get; set; } = string.Empty;
        public double Confidence { get; set; }
        public double AgreementScore { get; set; }
        public string MajorityPosition { get; set; } = string.Empty;
        public string? MinorityOpinion { get; set; }
        public string? DissentingAgentId { get; set; }
        public string KeyArgumentsJson { get; set; } = "[]";
        public string RisksJson { get; set; } = "[]";
        public string AssumptionsJson { get; set; } = "[]";
        public string RecommendedAction { get; set; } = string.Empty;
        public string DecisionMatrixJson { get; set; } = "{}";
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }

    public class KnowledgeDocument
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public string Title { get; set; } = string.Empty;
        public string FileType { get; set; } = "txt";
        public int ChunkCount { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }

    public class AuditLog
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public Guid? UserId { get; set; }
        public string Action { get; set; } = string.Empty;
        public string Details { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: backend/app/models/entities.py
import uuid
from datetime import datetime
from sqlalchemy import Column, String, Float, Integer, DateTime, ForeignKey, Text, JSON
from sqlalchemy.orm import relationship
from app.database import Base

def generate_uuid():
    return str(uuid.uuid4())

class User(Base):
    __tablename__ = "users"

    id = Column(String(36), primary_key=True, default=generate_uuid)
    email = Column(String(255), unique=True, index=True, nullable=False)
    password_hash = Column(String(255), nullable=False)
    full_name = Column(String(255), nullable=False)
    role = Column(String(50), default="Analyst")
    created_at = Column(DateTime, default=datetime.utcnow)

class AgentModel(Base):
    __tablename__ = "agents"

    id = Column(String(50), primary_key=True)
    name = Column(String(100), nullable=False)
    role = Column(String(100), nullable=False)
    description = Column(Text, nullable=False)
    icon = Column(String(20), nullable=False)

class DebateSession(Base):
    __tablename__ = "debate_sessions"

    id = Column(String(36), primary_key=True, default=generate_uuid)
    user_id = Column(String(36), ForeignKey("users.id"), nullable=True)
    question = Column(Text, nullable=False)
    context = Column(Text, default="")
    status = Column(String(50), default="Pending")
    consensus_score = Column(Float, nullable=True)
    confidence_score = Column(Float, nullable=True)
    agreement_level = Column(Float, nullable=True)
    duration_ms = Column(Integer, default=0)
    llm_provider = Column(String(50), default="OpenAI")
    created_at = Column(DateTime, default=datetime.utcnow)
    completed_at = Column(DateTime, nullable=True)

    opinions = relationship("AgentOpinion", back_populates="session", cascade="all, delete-orphan")
    debate_messages = relationship("DebateMessage", back_populates="session", cascade="all, delete-orphan")
    consensus_decision = relationship("ConsensusDecision", back_populates="session", uselist=False, cascade="all, delete-orphan")

class AgentOpinion(Base):
    __tablename__ = "agent_opinions"

    id = Column(String(36), primary_key=True, default=generate_uuid)
    debate_session_id = Column(String(36), ForeignKey("debate_sessions.id"), nullable=False)
    agent_id = Column(String(50), nullable=False)
    recommendation = Column(Text, nullable=False)
    reasoning_summary = Column(Text, nullable=False)
    confidence = Column(Float, default=0.0)
    risks_json = Column(JSON, default=list)
    assumptions_json = Column(JSON, default=list)
    counterarguments_json = Column(JSON, default=list)
    created_at = Column(DateTime, default=datetime.utcnow)

    session = relationship("DebateSession", back_populates="opinions")

class DebateMessage(Base):
    __tablename__ = "debate_messages"

    id = Column(String(36), primary_key=True, default=generate_uuid)
    debate_session_id = Column(String(36), ForeignKey("debate_sessions.id"), nullable=False)
    sender_agent_id = Column(String(50), nullable=False)
    target_agent_id = Column(String(50), nullable=True)
    message_type = Column(String(50), default="Critique")
    content = Column(Text, nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow)

    session = relationship("DebateSession", back_populates="debate_messages")

class ConsensusDecision(Base):
    __tablename__ = "consensus_decisions"

    id = Column(String(36), primary_key=True, default=generate_uuid)
    debate_session_id = Column(String(36), ForeignKey("debate_sessions.id"), nullable=False)
    final_recommendation = Column(Text, nullable=False)
    summary = Column(Text, nullable=False)
    confidence = Column(Float, default=0.0)
    agreement_score = Column(Float, default=0.0)
    majority_position = Column(Text, default="")
    minority_opinion = Column(Text, nullable=True)
    dissenting_agent_id = Column(String(50), nullable=True)
    key_arguments_json = Column(JSON, default=list)
    risks_json = Column(JSON, default=list)
    assumptions_json = Column(JSON, default=list)
    recommended_action = Column(Text, default="")
    decision_matrix_json = Column(JSON, default=dict)
    created_at = Column(DateTime, default=datetime.utcnow)

    session = relationship("DebateSession", back_populates="consensus_decision")

class KnowledgeDocument(Base):
    __tablename__ = "knowledge_documents"

    id = Column(String(36), primary_key=True, default=generate_uuid)
    title = Column(String(255), nullable=False)
    file_type = Column(String(50), default="txt")
    chunk_count = Column(Integer, default=0)
    created_at = Column(DateTime, default=datetime.utcnow)
